src/test/java/org/prebid/server/auction/ExchangeServiceTest.java

```diff
@@ -198,11 +199,30 @@ public void shouldTolerateImpWithUnknownBidderInExtension() {
 
         // then
         verify(bidderCatalog).isValidName(eq("invalid"));
-        verifyNoMoreInteractions(bidderCatalog);
+        verify(bidderCatalog).isDeprecatedName(eq("invalid"));
         verifyZeroInteractions(bidderRequester);
         assertThat(bidResponse).isNotNull();
     }
 
+    @Test
+    public void shouldProccesRequestAndAddErrorAboutDeprecatedBidder() {
+        String invalidBidderName = "invalid";
+
+        given(bidderCatalog.isValidName(invalidBidderName)).willReturn(false);
+        given(bidderCatalog.isDeprecatedName(invalidBidderName)).willReturn(true);
+        given(bidderCatalog.errorForDeprecatedName(invalidBidderName)).willReturn("invalid has been deprecated and is no longer available. Use valid instead.");
+
+        final BidRequest bidRequest = givenBidRequest(givenSingleImp(singletonMap(invalidBidderName, 0)));
+
+        final BidResponse bidResponse = exchangeService.holdAuction(bidRequest, uidsCookie, timeout, metricsContext).result();
+
+        Map<String, List<String>> errors = new HashMap<>();
+        errors.put(invalidBidderName, Collections.singletonList("invalid has been deprecated and is no longer available. Use valid instead."));
+
+        assertThat(bidResponse).hasFieldOrPropertyWithValue("ext",
```

Comment:
Please avoid using this feature as it kills type safety, it could be safely replaced with `assertThat(bidResponse.getExt()).equals(mapper...)`